crates/astria-sequencer-relayer/tests/blackbox/helper.rs

```diff
@@ -240,75 +253,71 @@ impl MockCelestia {
+impl HeaderServer for HeaderServerImpl {
+    async fn header_network_head(
+        &self,
+    ) -> Result<celestia_client::celestia_types::ExtendedHeader, ErrorObjectOwned> {
+        use celestia_client::{
+            celestia_tendermint::{
+                block::{
+                    header::Header,
+                    Commit,
+                },
+                validator,
+            },
+            celestia_types::{
+                DataAvailabilityHeader,
+                ExtendedHeader,
+            },
         };
-        use serde_json::{
-            to_string,
-            value::RawValue,
-            Value,
+        let header = ExtendedHeader {
+            header: Header {
+                height: 42u32.into(),
+                ..make_celestia_tendermint_header()
+            },
+            commit: Commit {
+                height: 42u32.into(),
+                ..Commit::default()
+            },
+            validator_set: validator::Set::without_proposer(vec![]),
+            dah: DataAvailabilityHeader {
+                row_roots: vec![],
+                column_roots: vec![],
+            },
         };
-        let rsp = RawValue::from_string(
-            to_string(&NetworkHeaderResponse {
-                commit: Commit {
-                    height: 42,
-                    rest: Value::default(),
-                },
-                inner: Value::default(),
-            })
-            .unwrap(),
-        )
-        .unwrap();
-        Ok(rsp)
+        Ok(header)
     }
 }
 
-struct StateCelestiaImpl {
+struct BlobServerImpl {
     sequencer_block_time_ms: u64,
     mode: CelestiaMode,
     rpc_confirmed_tx: mpsc::UnboundedSender<Vec<Blob>>,
 }
 
 #[async_trait]
-impl StateServer for StateCelestiaImpl {
-    async fn submit_pay_for_blob(
+impl BlobServer for BlobServerImpl {
+    async fn blob_submit(
         &self,
-        _fee: Fee,
-        _gas_limit: u64,
         blobs: Vec<Blob>,
-    ) -> Result<Box<serde_json::value::RawValue>, ErrorObjectOwned> {
-        use astria_celestia_jsonrpc_client::state::SubmitPayForBlobResponse;
-        use serde_json::{
-            to_string,
-            value::RawValue,
-            Value,
-        };
-
+        _opts: SubmitOptions,
+    ) -> Result<u64, ErrorObjectOwned> {
         self.rpc_confirmed_tx.send(blobs).unwrap();
-
-        let rsp = RawValue::from_string(
-            to_string(&SubmitPayForBlobResponse {
-                height: 100,
-                rest: Value::Null,
-            })
-            .unwrap(),
-        )
-        .unwrap();
         if let CelestiaMode::Delayed(n) = self.mode {
             tokio::time::sleep(Duration::from_millis(n * self.sequencer_block_time_ms)).await;
         }
-
-        Ok(rsp)
+        Ok(100)
     }
 }
 
-fn create_block_response(validator: &Validator, height: u32) -> endpoint::block::Response {
+fn create_block_response(
+    signing_key: &SigningKey,
+    account: tendermint::account::Id,
```

Comment:
derive the account ID from the signing key in here instead of passing it in every time? cause it appears the same thing is passed each time